src/config-loader.ts

```diff
@@ -0,0 +1,16 @@
+/**
+ * Enumerate the kinds of results we can get from a configuration loading source.
+ */
+export enum ConfigLoaderStatus {
+  FAILED, // An error was encountered
+  DID_NOT_PRODUCE, // There was no error encountered, but other logic resulted in no useful configuration from this source
+  COMPLETED, // Useful configuration was successfully loaded into the configuration store from this source.
+}
+
+export enum ConfigSource {
+  CONFIG_STORE,
+  FETCH,
+  NONE,
+}
+
+export type ConfigurationLoadAttempt = Promise<[ConfigSource, ConfigLoaderStatus]>;
```

Comment:
Typically, I'd expect to see this as an object (as opposed to an array) as it would make accessing the two fields a bit more explicit, but I don't have a strong preference either way